lib/metrics/processResources.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+const Gauge = require('../gauge');
+const { updateMetrics } = require('./helpers/processMetricsHelpers');
+
+const NODEJS_ACTIVE_RESOURCES = 'nodejs_active_resources';
+const NODEJS_ACTIVE_RESOURCES_TOTAL = 'nodejs_active_resources_total';
+
+module.exports = (registry, config = {}) => {
+	// Don't do anything if the function does not exist in previous nodes (exists in node@17.3.0)
+	if (typeof process.getActiveResourcesInfo !== 'function') {
+		return;
+	}
+
+	const namePrefix = config.prefix ? config.prefix : '';
+	const labels = config.labels ? config.labels : {};
+	const labelNames = Object.keys(labels);
+
+	new Gauge({
+		name: namePrefix + NODEJS_ACTIVE_RESOURCES,
+		help:
+			'Number of active resources that are currently keeping the event loop alive, grouped by async resource type.',
+		labelNames: ['type', ...labelNames],
+		registers: registry ? [registry] : undefined,
+		collect() {
+			const resources = process.getActiveResourcesInfo();
+
+			const data = {};
+
+			for (let i = 0; i < resources.length; i++) {
+				const resource = resources[i];
+
+				if (Object.hasOwnProperty.call(data, resource)) {
```

Comment:
Since this requires Node.js v17.3.0, you could use `Object.hasOwn(data, resource)` here.

Reply:
@zbjornson Thanks for the review! Updated accordingly, PTAL.